prac_05/emails.py

```diff
@@ -0,0 +1,53 @@
+"""Extract name from email, confirm user name, print name and email."""
+
+# Estimated time: 1.5 hours
+# Actual time: 4 hours
+
+SPECIAL_CHARACTERS = '@'
+
+
+def main():
+    """Extract name from email, confirm user name, print name and email."""
+    emails_and_names = {}
+
+    email = input("Email: ").strip()
+    email = get_valid_email(email)
+    while email != '':
+        name = get_name_from_email(email)
+        is_name_correct = input(f"Is this your name {name}? (Y/n): ").strip().lower()
+        if is_name_correct == 'n' or is_name_correct == 'no':
```

Comment:
In this case, we want to prompt for the name if the input is anything other than "" or "y". An if-else statement might be more suitable for this purpose.